### PR TITLE
excel: support for formula type cells

### PR DESCRIPTION
**Pull request by asce4s**
Thanks for contributing.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

1. Changed the code to support data from excel formulas from issue : #854 

## Testing

1. Unit Test cases added 
2. columns.xlxs and columns-with-missing-values.xlsx updated with new data

### CODE REVIEW

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 1]
```diff
@@ -145,14 +138,16 @@ private Boolean isBlank(Cell cell) {
     return null;
   }
 
-  private ColumnType getColumnType(Cell cell) {
-    switch (cell.getCellType()) {
+  private ColumnType getColumnType(Cell cell,CellType cellType) {
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:141]:
```suggestion
  private ColumnType getColumnType(Cell cell, CellType cellType) {
```

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 2]
```diff
@@ -145,14 +138,16 @@ private Boolean isBlank(Cell cell) {
...
+    switch (cellType) {
       case STRING:
         return ColumnType.STRING;
       case NUMERIC:
         return DateUtil.isCellDateFormatted(cell) ? ColumnType.LOCAL_DATE_TIME : ColumnType.INTEGER;
       case BOOLEAN:
         return ColumnType.BOOLEAN;
+      case FORMULA:
+        return getColumnType(cell,cell.getCachedFormulaResultType());
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:150]:
```suggestion
        return getColumnType(cell, cell.getCachedFormulaResultType());
```

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 3]
```diff
@@ -269,7 +264,7 @@ private Table createTable(Sheet sheet, TableRange tableArea, XlsxReadOptions opt
               column.appendMissing();
             }
           }
-          Column<?> altColumn = appendValue(column, cell);
+          Column<?> altColumn = appendValue(column, cell,cell.getCellType());
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:267]:
```suggestion
          Column<?> altColumn = appendValue(column, cell, cell.getCellType());
```

**Comment by benmccann** [comment]
Please include a space after each comma to match the existing code style

File: excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java [hunk 4]
```diff
@@ -84,7 +82,8 @@ public void testColumns() {
             "longcol",
             "doublecol",
             "booleancol",
-            "datecol");
+            "datecol",
+                "formulacol");
```
Comment (benmccann) [inline excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java:86]: this line should align with the others

File: excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java [hunk 5]
```diff
@@ -98,6 +97,7 @@ public void testColumns() {
         table.dateTimeColumn("datecol"),
         LocalDateTime.of(2019, 2, 22, 20, 54, 9),
         LocalDateTime.of(2020, 3, 23, 21, 55, 10));
+    assertColumnValues(table.doubleColumn("formulacol"),135.34 ,137.35);
```
Comment (benmccann) [inline excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java:100]: this line should align with the others

File: excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java [hunk 6]
```diff
@@ -112,7 +112,8 @@ public void testColumnsWithMissingValues() {
             "longcol",
             "doublecol",
             "booleancol",
-            "datecol");
+            "datecol",
+                "formulacol");
```
Comment (benmccann) [inline excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java:116]: this line should align with the others

Comment (asce4s) [reply excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java]: fixed

File: excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java (hunk 5 above)
Comment (asce4s) [reply excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java]: fixed

File: excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java (hunk 4 above)
Comment (asce4s) [reply excel/src/test/java/tech/tablesaw/io/xlsx/XlsxReaderTest.java]: fixed

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java (hunk 3 above)
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:267]:
```suggestion
          Column<?> altColumn = appendValue(column, cell, cell.getCellType());
```

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 7]
```diff
@@ -352,7 +349,7 @@ private Table createTable(Sheet sheet, TableRange tableArea, XlsxReadOptions opt
 
   private Column<?> createColumn(String name, Cell cell) {
     Column<?> column;
-    ColumnType columnType = getColumnType(cell);
+    ColumnType columnType = getColumnType(cell,cell.getCellType());
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:352]:
```suggestion
    ColumnType columnType = getColumnType(cell, cell.getCellType());
```

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 8]
```diff
@@ -145,14 +138,16 @@ private Boolean isBlank(Cell cell) {
     return null;
   }
 
-  private ColumnType getColumnType(Cell cell) {
-    switch (cell.getCellType()) {
+  private ColumnType getColumnType(Cell cell, CellType cellType) {
+    switch (cellType) {
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:142]: I think I'd probably make the first line of this method get the cell type and leave this method with a single parameter

```
  private ColumnType getColumnType(Cell cell) {
      CellType cellType = cell.getCellType() == FORMULA ? cell.getCachedFormulaResultType() : cell.getCellType();
      switch (cellType) {
```

And do the same for `appendValue` below

Comment (asce4s) [reply excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:145]: made the changes

File: excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java [hunk 9]
```diff
@@ -146,13 +141,16 @@ private Boolean isBlank(Cell cell) {
...
-    switch (cell.getCellType()) {
+    CellType cellType = cell.getCellType() == FORMULA ? cell.getCachedFormulaResultType() : cell.getCellType();
+    switch (cellType) {
       case STRING:
         return ColumnType.STRING;
       case NUMERIC:
         return DateUtil.isCellDateFormatted(cell) ? ColumnType.LOCAL_DATE_TIME : ColumnType.INTEGER;
       case BOOLEAN:
         return ColumnType.BOOLEAN;
+      case FORMULA:
```
Comment (benmccann) [inline excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java:152]: I don't think you still need this `case` anymore (same below in `appendValue`)

Comment (asce4s) [reply excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java]: yes..I missed that

**Pull request merged**